code/lib/core-events/src/index.ts

```diff
@@ -75,6 +75,8 @@ enum events {
   TELEMETRY_ERROR = 'telemetryError',
   FILE_COMPONENT_SEARCH = 'fileComponentSearch',
   FILE_COMPONENT_SEARCH_RESULT = 'fileComponentSearchResult',
+  CREATE_NEW_STORYFILE = 'createNewStoryfile',
+  CREATE_NEW_STORYFILE_RESULT = 'createNewStoryfileResult',
```

Comment:
I personally would treat them separately, because technically two completely different things happen. 

For creating a new story file, we don't have to take into account a base story or the structure of an existing story file. We can simply use two different kind of templates (one for JavaScript and one for TypeScript) to create these files.

On the other side, when creating a new story for an exisiting story file, we don't have to take care of creating a story files.

Therefore, I would like to rather listen to two different events (they also get triggered differently in the UI) than to do some payload checking to apply one or the other logic.

Reply:
I agree with @valentinpalkovic, the objectives seem different enough to warrant 2 different core-events